src/main/java/teammates/common/util/Assumption.java

```diff
@@ -288,6 +288,9 @@ public static void assertNull(String message, Object object) {
      * Asserts that two objects refer to the same object. If they are not an
      * AssertionFailedError is thrown with the given message.
      */
+    @SuppressWarnings("PMD.CompareObjectsWithEquals")
+    // PMD.CompareObjectsWithEquals is suppressed as assertSame are checking if 
+    // the expected and actual are the same objects 
```

Comment:
Then this method is no longer assertSame then, but assertEquals. There's no side effect to removing this entirely though, since there isn't any location in the code that uses this.